### PR TITLE
Add test for DROP TABLE IF EXISTS statement with non-existing object

### PR DESCRIPTION
**Pull request by krvikash**
## Description
Add test for DROP TABLE IF EXISTS statement with non-existing object

## Additional context and related issues

NA

## Release notes

(X) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by findinpath — APPROVED** [review]

**Comment by findepi** [comment]
@krvikash  there  was a force push to the  master branch (per https://github.com/trinodb/trino/pull/15365#issuecomment-1358243396), so your PR appears as if containing some unrelated commits.
Can you please rebase to remove them?

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 1]
```diff
@@ -5030,6 +5030,14 @@ public void testMergeAllColumnsReversed()
         assertUpdate("DROP TABLE " + targetTable);
     }
 
+    @Test
+    public void testDropTableIfExistsStatementWithNonExistingObject()
+    {
+        assertUpdate("DROP TABLE IF EXISTS non_existing_catalog.non_existing_schema.non_existing_table");
+        assertUpdate("DROP TABLE IF EXISTS non_existing_schema.non_existing_table");
+        assertUpdate("DROP TABLE IF EXISTS non_existing_table");
```
Comment (ebyhr) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:5038]: `testDropTableIfExists` already covers this one.

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 2]
```diff
@@ -5030,6 +5030,14 @@ public void testMergeAllColumnsReversed()
         assertUpdate("DROP TABLE " + targetTable);
     }
 
+    @Test
+    public void testDropTableIfExistsStatementWithNonExistingObject()
+    {
+        assertUpdate("DROP TABLE IF EXISTS non_existing_catalog.non_existing_schema.non_existing_table");
```
[the remark this answers is not in the document]
Comment (krvikash) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:5036]: I saw `testDropTable` and `testDropTableIfExists` tests in `BaseConnectorTest`, so I thought of adding this test in here.

In https://github.com/trinodb/trino/pull/15065, With the current approach, we are going to verify `tableExists` (which eventually calls `ConnectorMetadata.tableExists`)  in `DropTableTask` before we verify the statement.

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java (hunk 1 above)
Comment (krvikash) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:5038]: Is that okay to merge all statements in `testDropTableIfExists` or we can have separate tests for non-existing catalog and schema?

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (krvikash) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:5036]: @findepi, Please suggest if we can move this test case to some other class like `TestDropTableTask`?

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java (hunk 1 above)
Comment (ebyhr) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:5038]: > for non-existing catalog and schema

No need to add non-existing catalog case to BaseConnectorTest in my opinion. You can just add non-existing schema case to `testDropTableIfExists`. A separated test method is also acceptable, but the place should be closed to `testDropTableIfExists`.

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (findepi) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:5036]: `TestDropTableTask` sounds good to me

[the remark this answers is not in the document]
Comment (krvikash) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java]: Thanks! moved test cases to `TestDropTableTask`

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java (hunk 1 above)
Comment (krvikash) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java]: Moved test cases to `TestDropTableTask`. Please see https://github.com/trinodb/trino/pull/15464#discussion_r1054386010

File: core/trino-main/src/test/java/io/trino/execution/TestDropTableTask.java [hunk 3]
```diff
@@ -63,6 +63,24 @@ public void testDropNotExistingTableIfExists()
         // no exception
     }
 
+    @Test
+    public void testDropNotExistingSchemaIfExists()
```
Comment (ebyhr) [inline core/trino-main/src/test/java/io/trino/execution/TestDropTableTask.java:67]: "testDropNotExistingSchema..." sounds like dropping a schema. How about `testDropTableIfExistsWithoutExistingSchema` or something?

File: core/trino-main/src/test/java/io/trino/execution/TestDropTableTask.java [hunk 4]
```diff
@@ -63,6 +63,24 @@ public void testDropNotExistingTableIfExists()
...
+    {
+        QualifiedName tableName = QualifiedName.of(TEST_CATALOG_NAME, "non_existing_schema", "not_existing_table");
+
+        getFutureValue(executeDropTable(tableName, true));
+        // no exception
+    }
+
+    @Test
+    public void testDropNotExistingCatalogIfExists()
```
Comment (ebyhr) [inline core/trino-main/src/test/java/io/trino/execution/TestDropTableTask.java:76]: ditto

File: core/trino-main/src/test/java/io/trino/execution/TestDropTableTask.java [hunk 5]
```diff
@@ -63,6 +63,24 @@ public void testDropNotExistingTableIfExists()
...
+    {
+        QualifiedName tableName = QualifiedName.of("non_existing_catalog", "non_existing_schema", "not_existing_table");
+
+        getFutureValue(executeDropTable(tableName, true));
+        // no exception
+    }
```
Comment (ebyhr) [inline core/trino-main/src/test/java/io/trino/execution/TestDropTableTask.java:82]: `testDropNotExistingCatalogIfExists` → `testDropNotExistingSchemaIfExists` → `testDropNotExistingTableIfExists` is the natural order to me.

**Review by ebyhr — APPROVED** [review]

File: core/trino-main/src/test/java/io/trino/execution/TestDropTableTask.java (hunk 3 above)
Comment (krvikash) [reply core/trino-main/src/test/java/io/trino/execution/TestDropTableTask.java:67]: Yes sounds good. will make the changes.

**Comment by ebyhr** [comment]
CI hit #15367

**Pull request merged**